server/routes/main.js

```diff
@@ -0,0 +1,262 @@
+import { Router } from 'express';
+import faker from 'faker';
+import Product, { reviewSchema, productSchema } from '../models/product.js';
+
+const router = Router();
+
+router.get('/generate-fake-data', (req, res, next) => {
+  for (let i = 0; i < 90; i += 1) {
+    const product = new Product();
+    const fakeStarterReview = {
+      userName: faker.internet.userName(),
+      text: 'Fake review text',
+    };
+
+    product.category = faker.commerce.department();
+    product.name = faker.commerce.productName();
+    product.price = faker.commerce.price();
+    product.image = 'https://via.placeholder.com/250?text=Product+Image';
+    product.reviews = [fakeStarterReview];
+
+    product.save().then((__, err) => {
+      if (err) {
+        throw err;
+      }
+    });
+  }
+  res.end();
+});
+
+router.get('/products', (req, res, next) => {
+  const perPage = 9;
+  let { page = 1, category, price, query } = req.query;
+
+  const dbQuery = Product.find();
+
+  if (category) {
+    dbQuery.find({ category });
+  }
+
+  if (price) {
+    if (price === 'highest') {
```

Comment:
Move these words to a const, so you dont repeat and have a clear definition, in typescript you would use an enum